src/unix/core.c

```diff
@@ -1440,7 +1440,22 @@ int uv_gettimeofday(uv_timeval_t* tv) {
   if (gettimeofday(&time, NULL) != 0)
     return UV__ERR(errno);
 
-  tv->tv_sec = time.tv_sec;
-  tv->tv_usec = time.tv_usec;
+  tv->tv_sec = (int64_t)time.tv_sec;
+  tv->tv_nsec = (int64_t)time.tv_usec * 1000;
```

Comment:
Can we stick with `usec` to make it more of a drop in replacement, and so that users aren't confused?

Reply:
But then it doesn't make sense to use 64bit int for a value that should be < 1'000'000

Reply:
Maybe not, but making this nanoseconds seems like a sure fire footgun for anyone using this that is familiar with `gettimeofday()`.